### PR TITLE
RichText: Remove empty link formats during parse and edit.

### LINKED ISSUE #81936: Accessibility: Visual highlight for empty links in the editor
## What problem does this address?
When doing accessibility remediation, one of the most common issues that we see are empty links hidden within content. In the block editor, users can accidentally create empty links without fully removing the `<a>` wrapper when they edit or remove existing links. They may also paste in content for other places that include empty links.

These links are invisible in the visual editor and impossible to edit. When a link within text (like a heading, paragraph, or list block) is empty, it is 0 pixels wide, which means it cannot be clicked and edited or removed as can other links. The only fix for these links is to switch to the code editor and delete the link code; this is not intutive or user-friendly for non-technical people who don't know HTML.

### Example screenshots
This screenshot shows the block editor as it is currently. The Accessibility Checker plugin is active and has found 4 empty links, but nothing in the editor content helps users to see where those four links might be.
Accessibility Checker open in the block sidebar showing there are 4 empty links present on the page, a heading block, paragraph blocks, and list blocks with filler content are present and only one visible link is present.

Using Accessibility Checker, users can view the link on the front end and it will be highlighted.
Accessibility Checker showing an empty link in a heading block. 4 problem icons are visible on the page.

Users should not have to go to the front end to view problems like these.

### Why it matters
Empty links such as these create unneccessary tab stops for keyboard users, and are announced by screen readers simply as "link". Links such as these cause both WCAG failures and confusion for people with disabilities.

They are also just a general problem with content quality or, if excessively present, could potentially make a page look suspicious or negatively impact SEO.

Not being able to edit or remove links like this in visual mode is frustrating to non-technical users.

## Code to reproduce
To see the current experience, paste this code in the block editor:
```
<!-- wp:heading -->
<h2 class="wp-block-heading">This is a heading<a href="https://google.com"></a></h2>
<!-- /wp:heading -->

<!-- wp:paragraph -->
<p>This is a paragraph<a href="https://google.com"></a>. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>
<!-- /wp:paragraph -->

<!-- wp:paragraph -->
<p>Here's a list:</p>
<!-- /wp:paragraph -->

<!-- wp:list -->
<ul class="wp-block-list"><!-- wp:list-item -->
<li><a href="https://google.com"></a>list item 1</li>
<!-- /wp:list-item -->

<!-- wp:list-item -->
<li>list item 2</li>
<!-- /wp:list-item -->

<!-- wp:list-item -->
<li>list item 3</li>
<!-- /wp:list-item --></ul>
<!-- /wp:list -->

<!-- wp:heading {"level":3} -->
<h3 class="wp-block-heading">Real world example:</h3>
<!-- /wp:heading -->

<!-- wp:paragraph -->
<p>Contact us at <a href="mailto:hello@example.com"></a><a href="mailto:hello@example.com">hello@example.com</a>.</p>
<!-- /wp:paragraph -->
```

## What is your proposed solution?

Here is how links look while being edited.
Block editor showing an email link being edited. While being edited, the link has a gray background.

I proposed that empty links should always have:
1. At least 12px of left and right padding so they have a clickable area of 24px.
2. A visible red background so they standout as a possible problem when editing the page.

These styles should always be visible even if you are not currently editing the block.

Clicking on a link would allow you to insert text or remove it.

Here is an example of block editor styles that clearly highlight the empty links:
24 pixel red boxes appear within or at the end of a heading, list items, and paragraph blocks.

### PR DESCRIPTION
**Pull request by dilipom13**
Fixes #81936

## What?

Automatically removes empty link formats from rich text content during parsing and editing, instead of preserving invisible zero-width anchors in the block editor.

## Why?

Empty `<a href="..."></a>` elements were stored as object replacements and serialized back to HTML, creating inaccessible links that users could not see or edit in visual mode. Maintainers agreed on auto-cleanup rather than visual highlighting.

## How?

- Skip promoting empty anchor tags to object replacements in `createFromElement()`.
- Add `removeEmptyLinkFormats()` to strip any remaining empty link object replacements.
- Run cleanup when parsing HTML and before serializing edited values in the RichText hook.
- Add unit tests and a create spec for empty anchor tags.

## Testing Instructions

1. Paste the HTML repro from #81936 into a Paragraph or Heading block in the code editor.
2. Switch to visual editor, empty links should not be present.
3. Inspect serialized content, no `<a href="..."></a>` without text.
4. Confirm links with inline images still work.

Unit tests:
`npm run test:unit -- packages/rich-text/src/test/remove-empty-link-formats.js`

### CODE REVIEW

File: packages/rich-text/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+### Bug Fixes
+
+-   Remove empty link formats during RichText parsing and editing so invalid anchors are cleaned up automatically ([#81936](https://github.com/WordPress/gutenberg/issues/81936)).
```
Comment (shail-mehta) [inline packages/rich-text/CHANGELOG.md]: I think we should add the PR link here instead of the issue link.

Comment (dilipom13) [reply packages/rich-text/CHANGELOG.md]: Okay, @Sahil1617, I added this text as per your suggestion.
Thanks for this information.
